### PR TITLE
Prompt reconnect to AP when timeout during Wi-Fi setup

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/OS-1206) |

#### Main changes

On the `ConnectDialog` in the Wifi page, added a timeout to request that connects to a wifi network.

If the request timeouts:
- The regular "couldn't connect to network" error message is displayed.
- If we determine that the user was onboarding using AP mode, we also prompt them to reconnect to the 'pi-top' network.
- The app will continuously check if the connection succeeded. If it finds that it actually did, the dialog is updated to the "success" state.

#### Screenshots (feature, test output, profiling, dev tools etc)

image

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)

Since our server only responds to one request at the same time, the request that sets the credentials (`connectToNetwork` service) will block it. 

This means that even though we query the server to determine if we connected successfully with the `connectedBSSID` service, this request won't actually be processed until the other one finishes.

#### Manual testing tips

- Download and scp the deb files into your pi-top.
- Install the deb with `sudo apt install ./*.deb`.
- Restart the web-portal service with `sudo restart pt-os-web-portal`.
- Connect to AP mode.
- Go to the wifi page. Try to connect to a network with the wrong credentials. You should see two error messages, one telling you about the credentials and potentially the one telling you to reconnect to AP.
- The "AP" error should go away after a bit, once the network is detected again.
- Now again, try connecting to a network using the correct password. 

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: frontend/src/pages/wifiPage/connectDialog/ConnectDialog.tsx [hunk 1]
```diff
@@ -102,6 +104,12 @@ export default ({
           <span className={styles.error}>
             There was an error connecting to {ssid}... please check your
             password and try again
+            {disconnectedFromAp && (
+              <>
+              <br></br>
+              Your computer has disconnected from the wifi network 'pi-top'. Please reconnect to it.
+              </>
+            )}
```
Comment (olivierwilkinson) [inline frontend/src/pages/wifiPage/connectDialog/ConnectDialog.tsx:112]: When disconnectedFromAp is true we don't actually know that connecting to the ssid was unsuccessful. Plus having both messages at once is a lot of text.

```suggestion
          <span className={styles.error}>
            {disconnectedFromAp ? (
              <>
                Your computer has disconnected from the wifi network
                'pi-top-XXX'. Please reconnect to it.
              </>
            ) : (
              <>
                There was an error connecting to {ssid}... please check your
                password and try again
              </>
            )}
```

**Review by angusjfw — APPROVED** [review]
